src/assets/toolkit/scripts/lib/component/slideshow.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+import {u} from 'umbrellajs';
+export class Slideshow {
+  constructor (element, {
+    slideHolder = element.querySelector('.js-Slideshow-slides'),
+    slides = element.querySelectorAll('.js-Slideshow-slide'),
+    nextTrigger = element.querySelector('.js-Slideshow-next'),
+    prevTrigger = element.querySelector('.js-Slideshow-prev'),
+    currentCountElement = element.querySelector('.js-Slideshow-current'),
+    totalCountElement = element.querySelector('.js-Slideshow-total'),
+    classIsVisible = 'is-visible',
+    classWasVisible = 'was-visible',
+    classIsSlidingForward = 'is-sliding-forward',
+    classIsSlidingBack = 'is-sliding-back',
+    classIsForward = 'is-forward',
+    classIsBack = 'is-back'
+  } = {}) {
+
+    Object.assign(this, {
+      slides,
+      slideHolder,
+      nextTrigger,
+      prevTrigger,
+      currentCountElement,
+      totalCountElement,
+      classIsVisible,
+      classWasVisible,
+      classIsForward,
+      classIsBack,
+      classIsSlidingForward,
+      classIsSlidingBack
+    });
+
+    this.counter = 0;
+    u(this.totalCountElement).text(this.numSlides);
+    u(this.nextTrigger).handle('click', () => this.next());
+    u(this.prevTrigger).handle('click', () => this.prev());
+  }
+
+  get numSlides() {
+    return this.slides.length;
+  }
+
+  get slideToShow() {
+    var slideToShow;
+    var slideRemainder = this.counter % this.numSlides;
+
+    // Get the index of the current item
+    if (this.counter > 0) {
+      slideToShow = slideRemainder;
+    } else {
+      slideToShow = slideRemainder ? (slideRemainder + this.numSlides) : 0;
+    }
+
+    return slideToShow;
+  }
+
+  slide(direction) {
+
+    // Update the navigation with the current slide number
+    u(this.currentCountElement).text(this.slideToShow + 1);
+
+    // Remove previous directional class
+    u(this.slideHolder).removeClass(this.classIsForward, this.classIsBack);
+
+    // Remove previous 'was-visible' class
+    this.slides.forEach(slide => {
```

Comment:
Does Umbrella allow jQuery-style manipulation of en entire collection at once?

If so, could this be simplified to the following?

```js
this.slides.removeClass(this.classWasVisible);
```